proj/funcDictionaryJaccard.py

```diff
@@ -0,0 +1,139 @@
+import networkx as nx
+
+from imnet import process_strings as p
+
+import textdistance as td # better cite that
+
+from plotData import InnerDataLevenshtein
+
+import funcDictionary as dic
+
+# Check, if the description correspond to the functions
+
+def make_sub_graphs(G):
+    """
+    info: make list of connected subgraphs of G (subgraph with the highest number of nodes) 
+    input: G: graph
+    output: sub_graphs: list of subgraphs, sorted according to size
+    """ 
+
+    # info: make sorted list of subgraphs
+    sub_graphs = sorted(list(G.subgraph(c) for c in nx.connected_components(G)), key = len, reverse = True)
+    
+    # info: if sub_graphs is empty , make a base graph
+    if len(sub_graphs) == 0:
+        sub_graphs = list()
+        sub_graphs.append(nx.Graph())
+        sub_graphs[0].add_node('FillNode')
+
+    return sub_graphs
+
+def degree_mean(GSub):
+    """
+    info: calculate the average degree in the graph GSub
+    input: GSub: graph
+    output: deg: average degree number (real number)
+    """
+    deg = 0
+    degs = GSub.degree()
+    degs = list(degs)
+    for el in degs:
+        deg += el[1]/(len(degs))
+    return deg
+
+def makeNet(names, maxDist):
+    """
+    info: make graph, where each node denotes one element of names. 
+        names is an array of strings.
+        if the jaccard distance between the strings, corresponding 
+        to two nodes, is lower than "maxDist", an edge is drawn
+    input: names:[String]:  list of strings
+        maxDist (real number from 0 to 1): maximum jaccard distance between two strings,
+            which is required to form a bond
+    output: g: created graph, where each strain is a node and two nodes are
+        connected via an edge, if there jaccard distance is smaller
+        than maxDist
+    """
+    g = nx.Graph()
+
+    #print("\n names: ", names)
+    for i in range(len(names)):
+        g.add_node(names[i])
+
+    for i in range(len(names)):
```

Comment:
please use list comprehension and not a loop or use a loop if you want to parallelize later on. it really depends on what you want to do here. you can use `numba` or `pyspark`. 

https://wiki.python.org/moin/ParallelProcessing